src/sidebar/util/oauth-client.js

```diff
@@ -182,19 +180,15 @@ export default class OAuthClient {
     });
 
     // Authorize user and retrieve grant token
-    let authUrl = this.authorizationEndpoint;
-    authUrl +=
-      '?' +
-      queryString.stringify({
-        client_id: this.clientId,
-        origin: $window.location.origin,
-        response_mode: 'web_message',
-        response_type: 'code',
-        state,
-      });
+    const authURL = new URL(this.authorizationEndpoint);
+    authURL.searchParams.set('client_id', this.clientId);
+    authURL.searchParams.set('origin', $window.location.origin);
+    authURL.searchParams.set('response_mode', 'web_message');
+    authURL.searchParams.set('response_type', 'code');
+    authURL.searchParams.set('state', state);
 
     // @ts-ignore - TS doesn't know about `location = <string>`.
```

Comment:
I'm curious about this. How does TS complain here?

Reply:
From memory it says something like `` `string` is not assignable to `Location` ``. The `window.location` is one of those legacy DOM properties that has some unusual behaviors.